HMCLCore/src/main/java/org/jackhuang/hmcl/util/platform/JavaVersion.java

```diff
@@ -369,18 +357,40 @@ private static Stream<Path> searchPotentialJavaExecutables() throws IOException
                         .map(JavaVersion::getExecutable));
                 javaExecutables.add(Stream.of(Paths.get("/Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/bin/java")));
                 javaExecutables.add(Stream.of(Paths.get("/Applications/Xcode.app/Contents/Applications/Application Loader.app/Contents/MacOS/itms/java/bin/java")));
-                javaExecutables.add(Stream.of(Paths.get("/Library/Application Support/minecraft/runtime/jre-x64/jre.bundle/Contents/Home/bin/java")));
-                if (System.getenv("PATH") != null) {
-                    javaExecutables.add(Arrays.stream(System.getenv("PATH").split(":")).flatMap(path -> Lang.toStream(FileUtils.tryGetPath(path, "java"))));
-                }
-                if (System.getenv("HMCL_JRES") != null) {
-                    javaExecutables.add(Arrays.stream(System.getenv("HMCL_JRES").split(":")).flatMap(path -> Lang.toStream(FileUtils.tryGetPath(path, "bin", "java"))));
-                }
-                break;
 
+                runtimeDirs.add(FileUtils.tryGetPath("/Library/Application Support/minecraft/runtime"));
+                runtimeDirs.add(FileUtils.tryGetPath(System.getProperty("user.home"), "/Library/Application Support/minecraft/runtime"));
+
+                runtimeOSArch.add("mac-os");
+                break;
+    
             default:
                 break;
         }
+
+        // Do MC runtimes, given the OS-specific info we have.
+        for (Optional<Path> runtimeDir : runtimeDirs) {
+            if (!runtimeDir.isPresent())
+                continue;
+            
+            for (String osArch : runtimeOSArch) {
+                javaExecutables.add(Stream.of(
+                    runtimeDir.get().resolve("jre-legacy").resolve(osArch).resolve("jre-legacy")),
+                    runtimeDir.get().resolve("java-runtime-alpha").resolve(osArch).resolve("jre-runtime-alpha"),
+                    runtimeDir.get().resolve("java-runtime-beta").resolve(osArch).resolve("jre-runtime-beta")
+                ).map(JavaVersion::getExecutable);
```

Comment:
not gonna work on a Mac due to bundle magic. Maybe we should specialize `getExecutable` for mac